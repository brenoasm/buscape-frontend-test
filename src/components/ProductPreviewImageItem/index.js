import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { colors } from "../../theme";

const propTypes = {};

const defaultProps = {};

const StyledProductPreviewImageItem = styled.div`
  display: flex;
  flex-direction: column;
  border: 1px solid
    ${props => (props.active ? colors.toolbarPrimary : colors.grayPrimary)};
  border-radius: 3px;
  padding: 3px;
  margin: 3px;

  > img {
    height: 40px;
    width: 40px;
  }
`;

const ProductPreviewImageItem = ({
  url,
  productId,
  active,
  onBrokenImageUrl,
  onSelectedImageChange
}) => {
  return (
    <StyledProductPreviewImageItem
      active={active}
      onClick={() => onSelectedImageChange(productId, url)}
    >
      <img
        onError={event => onBrokenImageUrl(event)}
        src={url}
        alt="Imagem do Produto"
      />
    </StyledProductPreviewImageItem>
  );
};

ProductPreviewImageItem.propTypes = propTypes;
ProductPreviewImageItem.defaultProps = defaultProps;

export default ProductPreviewImageItem;
