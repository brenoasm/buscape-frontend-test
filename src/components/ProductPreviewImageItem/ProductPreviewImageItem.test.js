import React from "react";
import { shallow } from "enzyme";

import ProductPreviewImageItem from "./";

describe("ProductPreviewImageItem", () => {
  const initialProps = {
    url:
      "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
    productId: 2321312
  };

  it("matches the snapshot", () => {
    const wrapper = shallow(<ProductPreviewImageItem {...initialProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
