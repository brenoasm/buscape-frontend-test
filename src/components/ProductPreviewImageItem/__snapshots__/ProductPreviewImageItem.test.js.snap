// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPreviewImageItem matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductPreviewImageItem
    active={false}
    onBrokenImageUrl={[Function]}
    onSelectedImageChange={[Function]}
    productId={2321312}
    url="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": false,
      "children": <Image
        alt="Produto"
        height={40}
        onBrokenImageUrl={[Function]}
        src="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
        width={40}
      />,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "alt": "Produto",
        "height": 40,
        "onBrokenImageUrl": [Function],
        "src": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
        "width": 40,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex-direction: column;
  border: 1px solid
    ",
          [Function],
          ";
  border-radius: 3px;
  padding: 3px;
  margin: 3px;
  cursor: pointer;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": false,
        "children": <Image
          alt="Produto"
          height={40}
          onBrokenImageUrl={[Function]}
          src="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
          width={40}
        />,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "Produto",
          "height": 40,
          "onBrokenImageUrl": [Function],
          "src": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
          "width": 40,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "rules": Array [
            "
  display: flex;
  flex-direction: column;
  border: 1px solid
    ",
            [Function],
            ";
  border-radius: 3px;
  padding: 3px;
  margin: 3px;
  cursor: pointer;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
